src/AndcultureCode.CSharp.Sitefinity.Core/JsonSerialization/Converters/SitefinityChoiceFieldEnumConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AndcultureCode.CSharp.Sitefinity.Core.JsonSerialization.Converters
+{
+    /// <summary>
+    /// Responsible for generically converting enums based on ChoiceFields into values that correspond with
+    /// how Sitefinity handles ChoiceField indices. See https://www.progress.com/documentation/sitefinity-cms/choicefield-operations
+    /// for more information on how the ChoiceField works through the OData web services.
+    ///
+    /// Developer Note: This converter operates under the assumption that the specified Enum has values starting
+    /// at 1 NOT 0.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class SitefinityChoiceFieldEnumConverter<T> : JsonConverter where T : struct, IConvertible
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (!typeof(T).IsEnum)
+            {
+                throw new ArgumentException("T must be an enumerated type");
+            }
+
+            long convertedValue = ((T) value).GetHashCode();
+
+            writer.WriteValue(Math.Pow(2, convertedValue - 1).ToString());
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (!typeof(T).IsEnum)
+            {
+                throw new ArgumentException("T must be an enumerated type");
+            }
+
+            if (reader.Value == null)
+            {
+                return null;
+            }
+
+            long convertedValue = Convert.ToInt64(reader.Value);
+
+            long sitefinityValue = (long) (Math.Log(convertedValue, 2) + 1);
+
+            return Enum.Parse(typeof(T), sitefinityValue.ToString());
```

Comment:
Same as above... consider `Enum.TryParse` if we decide to keep this converter based on the the outcome of the previous conversation with @brandongregoryscott